source/behaviors/offClick/offClick.ts

```diff
@@ -29,16 +30,36 @@ export class OffClickDirective implements OnInit, OnDestroy {
 	}
 
 	ngOnInit() {
-		setTimeout(() => {
-			document.addEventListener('click', this.listener);
-		}, 0);
+		if (this.active) {
+			setTimeout(() => {
+				this.addListener();
+			});
+		}
 	}
 
-	ngOnDestroy() {
+	ngOnChanges(changes: SimpleChanges): void {
```

Comment:
Actually @TheOriginalJosh it's not a null tri-state. That statement is checking to see if the 'changes' object has an 'active' change on it. And if there is an active change, we check for the boolean value. This is a must if we're using ngOnChanges at all, or else we'll just get null refs if it ever gets called when active didn't change.